src/main/java/pau/task/TaskList.java

```diff
@@ -0,0 +1,247 @@
+            int starEyesEmoji = 0x1F929;
+            System.out.println("yay you don't have anything to do" + new String(Character.toChars(starEyesEmoji)));
+        } else {
+            System.out.println("sian you still have to complete these:");
+            for (int i = 0; i < this.listSize(); i++) {
+                Task curr = this.taskList.get(i);
+                System.out.println((i + 1) + ". " + curr.toString());
+            }
+        }
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param input The task the user wants to mark as done.
+     */
+    public void markTask(String input) {
+        int starEyesEmoji = 0x1F929;
+        System.out.println("good job, you've completed a task! You're so productive!" + new String(Character.toChars(starEyesEmoji)));
+        String parts[] = input.split(" ");
+
+        int taskNo = Integer.parseInt(parts[1]);
+        Task checkedTask = this.taskList.get(taskNo - 1);
+
+        checkedTask.markAsDone();
+        System.out.println(checkedTask.toString());
+    }
+
+    /**
+     * Unmarks a task.
+     *
+     * @param input The task the user wants to unmark.
+     */
+    public void unMarkTask(String input) {
+        String parts[] = input.split(" ");
+        int taskNo = Integer.parseInt(parts[1]);
+        Task checkedTask = this.taskList.get(taskNo - 1);
+
+        checkedTask.markAsUndone();
+        System.out.println("why are you not going to " + checkedTask.description + "? remember to do it later!");
+        System.out.println(checkedTask.toString());
+    }
+
+    /**
+     * Deletes a Task from the list.
+     *
+     * @param input The task the user wants to delete.
+     */
+    public void deleteTask(String input) {
+        try {
+            String parts[] = input.split(" ");
+            int taskNo = Integer.parseInt(parts[1]);
+            if (taskNo > this.listSize()) {
+                throw new NoSuchTaskException("you can't delete this task");
+            }
+            Task checkedTask = this.taskList.get(taskNo - 1);
+            this.taskList.remove(checkedTask);
+            if (checkedTask.getStatusIcon().equals("X")) {
+                System.out.println("good job! you're officially done with this:");
+                System.out.println(checkedTask.toString());
+            } else {
+                System.out.println("not you running away from your responsibilities, i guess you don't have to do this now:");
+                System.out.println(checkedTask.toString());
+            }
+        } catch (NoSuchTaskException e) {
+        }
+
+        if (this.listSize() == 0) {
+            System.out.println("THERES NOTHING LEFT TO DO!!!!");
+        } else {
+            System.out.println("but still sucks to be you, you still have " + this.listSize() + " tasks");
+        }
+    }
+
+    /**
+     * Adds a ToDo to the task list.
+     *
+     * @param input The ToDo the user wants to add.
+     */
+    public void addToDo(String input) {
```

Comment:
Maybe instead of having an add method for each task type, you can have a general addtask method here, and abstract the specifics in each class (Todo, Deadline etc.)